File: Arrays_And_Hashing/0001-TwoSum/0001-hashmap-sol.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # Create Hashmap
        # loop through the list of integers
            # If complement is in hashmap, return i (index of current element) and index of complement in hashmap
            # Otherwise, add the index of the element into the hashmap
        
        hashmap = {}
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in hashmap:
                return [i,hashmap[complement]]
            else:
                hashmap[nums[i]] = i 
        return []

File: Arrays_And_Hashing/0001-TwoSum/0001-hashmap-sol.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> hashmap = new HashMap();
        int complement;
        for(int i = 0; i <nums.length; i++){
            complement = target - nums[i];
            if(hashmap.containsKey(complement)){
                return new int[]{i,hashmap.get(complement)};
            }
            else{
                hashmap.put(nums[i],i);
            }
        }
        return new int[]{};
    }
}